src/app/child-dev-project/notes/notes-migration/notes-migration.service.spec.ts

```diff
@@ -0,0 +1,150 @@
+import { TestBed } from "@angular/core/testing";
+
+import { NotesMigrationService } from "./notes-migration.service";
+import { EntityMapperService } from "../../../core/entity/entity-mapper.service";
+import { User } from "../../../core/user/user";
+import { Note } from "../model/note";
+import { mockEntityMapper } from "../../../core/entity/mock-entity-mapper-service";
+
+function legacyNote(author: string): Note {
+  const note = new Note();
+  note["author"] = author;
+  return note;
+}
+
+function createUser(name: string): User {
+  const user = new User();
+  user.name = name;
+  return user;
+}
+
+describe("NotesMigrationService", () => {
+  let service: NotesMigrationService;
+  const Peter = createUser("Peter");
+  const Ursula = createUser("Ursula");
+  const Jens = createUser("Jens");
+  const Angela = createUser("Angela");
+  const Albrecht = createUser("Albrecht");
+  const Johanna = createUser("Johanna");
+  const users = [Peter, Ursula, Jens, Angela, Albrecht, Johanna];
+
+  const entityMapper = mockEntityMapper();
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: EntityMapperService,
+          useValue: entityMapper,
+        },
+      ],
+    });
+    service = TestBed.inject(NotesMigrationService);
+    service.allUsers = new Map(users.map((u) => [u.name.toLowerCase(), u]));
+  });
+
+  it("should be created", () => {
+    expect(service).toBeTruthy();
+  });
+
+  it("finds the correct user for different user-names", () => {
+    const oldAuthors = [
+      "Peter Lustig, Ursula",
+      "Jens & Angela",
+      "Albrecht",
+      "Johanna",
+      "Peter, Jens, Albrecht, Johanna",
+    ];
+    const expectedUsers = [
+      [Peter, Ursula],
+      [Jens, Angela],
+      [Albrecht],
+      [Johanna],
+      [Peter, Jens, Albrecht, Johanna],
+    ];
+    oldAuthors.forEach((authorName, index) => {
+      const foundUsers = service.findUsers(authorName);
+      expect(foundUsers.detectedUsers).toEqual(expectedUsers[index]);
+      expect(foundUsers.additional).toHaveSize(0);
+    });
+  });
+
+  it("returns an array of users that could not be matched", () => {
+    const nonMatchedUsers = [
+      "Phillip",
+      "Christian",
+      "Agnes, Strack & Zimmermann",
+      "Gregor Giselle",
+      "Insalata and mista",
+    ];
+    const expected = [
+      ["Phillip"],
+      ["Christian"],
+      ["Agnes", "Strack", "Zimmermann"],
+      ["Gregor Giselle"],
+      ["Insalata", "mista"],
+    ];
+    nonMatchedUsers.forEach((userName, index) => {
+      const foundUsers = service.findUsers(userName);
+      expect(foundUsers.additional).toEqual(expected[index]);
+      expect(foundUsers.detectedUsers).toHaveSize(0);
+    });
+  });
+
+  it("migrates a note with existing users", () => {
+    const note = legacyNote("Peter L, Ursula & Jens");
+    const expectedUsers = users.slice(0, 3).map((u) => u.getId());
```

Comment:
```suggestion
    const expectedUsers = [Peter, Ursula, Jens].map(u => u.getId());
```
This would make the tests more readable